packages/react-reconciler/src/ReactChildFiber.new.js

```diff
@@ -293,6 +294,12 @@ function ChildReconciler(shouldTrackSideEffects) {
       returnFiber.deletions = [childToDelete];
       // TODO (effects) Rename this to better reflect its new usage (e.g. ChildDeletions)
       returnFiber.effectTag |= Deletion;
+
+      // We are deleting a subtree that may contain a passive effect.
+      // Mark the parent so we traverse this path after commit and run any unmount functions.
+      // This may cause us to traverse unnecessarily in some cases, but effects are common,
+      // and the cost of over traversing is small (just the path to the deleted node).
+      returnFiber.subtreeTag |= PassiveSubtreeTag;
```

Comment:
Could you instead add the Deletion bit to PassiveMask? Then you don't have to do anything extra in this function.

Reply:
Hm, yes, I think that would work. Sorry if I misinterpreted your original suggestion. I realize now that you were probably suggesting that instead of this.